File: src/driverdilemma.py
def cando(c, x, m, f, s):
    cost_hour = x + s / f
    time = m / s
    if time*cost_hour <= c/2: return True
    else: return False

if __name__ == "__main__":
    c, x, m = map(float, input().split())
    res = 0
    for i in range(6):
        s, f = map(float, input().split())
        if cando(c, x, m, f, s) == True: res = s
    
    if res == 0: print("NO")
    else: print(f"YES {int(res)}")

File: src/driverdilemma.cpp
#include<bits/stdc++.h>
using namespace std;

bool cando(double c, double x, double m, double f, double s){
    double cost_hour = x + s / f;
    double time = m / s;
    if(time * cost_hour <= c / 2) return true;
    else return false;
}

int main(){
    double c, x, m, f, res = 0, s;
    cin >> c >> x >> m;
    for(int i = 0; i < 6; i++){
        cin >> s >> f;
        if(cando(c, x, m, f, s) == true) res = s;
    }
    if(res == 0) cout << "NO" << endl;
    else cout << "YES " << res << endl; 
    return 0;
}